utils/data_helpers.py

```diff
@@ -204,26 +211,60 @@ def download_cub(include_text=False):
 
-def download_flowers():
+def download_flowers(include_text=False):
     """ Download the flowers dataset """
-    FLOWERS_DATASET_URL = "www.robots.ox.ac.uk/~vgg/data/flowers/102/102flowers.tgz"
+    FLOWERS_DATASET_URL = "http://www.robots.ox.ac.uk/~vgg/data/flowers/102/102flowers.tgz"
     print('Downloading the flowers dataset from: {}'.format(FLOWERS_DATASET_URL))
 
-    flowers_download_loc = pathlib.Path('data/flowers/flowers.tgz')
+    flowers_download_loc = pathlib.Path('data/flowers.tgz')
     urllib.request.urlretrieve(FLOWERS_DATASET_URL, flowers_download_loc)
     tar = tarfile.open(flowers_download_loc, "r:gz")
-    tar.extractall("data/flowers/images")
+    if include_text:
+        images_save_location = 'data/flowers_with_text/images/'
+    else:
+        images_save_location = 'data/flowers'
+    tar.extractall(images_save_location)
     tar.close()
     os.remove(flowers_download_loc)
 
-    DATA_SPLITS_URL = "www.robots.ox.ac.uk/~vgg/data/flowers/102/setid.mat"
-    data_splits_download_loc = pathlib.Path('data/flowers/setid.mat')
+    DATA_SPLITS_URL = "http://www.robots.ox.ac.uk/~vgg/data/flowers/102/setid.mat"
+    data_splits_download_loc = pathlib.Path(os.path.join(images_save_location, 'setid.mat'))
     urllib.request.urlretrieve(DATA_SPLITS_URL, data_splits_download_loc)
 
-    IMAGE_LABELS_URL = "www.robots.ox.ac.uk/~vgg/data/flowers/102/imagelabels.mat"
-    image_labels_download_loc = pathlib.Path('data/flowers/imagelabels.mat')
+    IMAGE_LABELS_URL = "http://www.robots.ox.ac.uk/~vgg/data/flowers/102/imagelabels.mat"
+    image_labels_download_loc = pathlib.Path(os.path.join(images_save_location, 'imagelabels.mat'))
     urllib.request.urlretrieve(IMAGE_LABELS_URL, image_labels_download_loc)
 
+    if include_text:
+        # Download the image captions
+        FLOWERS_TEXT_GDRIVE_ID = '0B3y_msrWZaXLaUc0UXpmcnhaVmM'
+
+        flowers_text_download_location = "data/flowers.zip"
+        flowers_text_backup_location = 'data/backup/flowers.zip'
+        extracted_text_dir = flowers_text_download_location[:-4]
+
+        if os.path.exists(flowers_text_backup_location):
+            # Restore from back up
+            print('Retrieving flowers dataset from: {}'.format(flowers_text_backup_location))
+            shutil.copy(flowers_text_backup_location, flowers_text_download_location)
+            with zipfile.ZipFile(flowers_text_backup_location, 'r') as zipfd:
+                zipfd.extractall('data/')
+        else:
+            # Clean download
+            print('Downloading flowers text from Google Drive ID: {}'.format(FLOWERS_TEXT_GDRIVE_ID))
+            gdd.download_file_from_google_drive(file_id=FLOWERS_TEXT_GDRIVE_ID,
+                                                dest_path=flowers_text_download_location,
+                                                unzip=True)
+            mkdir('data/backup')
+            shutil.copy(flowers_text_download_location, flowers_text_backup_location)
```

Comment:
we should definitely make this backup functionality a function but not sure how straight forward that is, might be too much custom code per use case

Reply:
Done :+1: